tools/ci/verify_bundled_optional.sh

```diff
@@ -17,24 +17,51 @@
 # limitations under the License.
 #
 
+usage() {
+  echo "Usage: $0 <maven-build-output>"
+  echo "    <maven-build-output>                 A file containing the output of the Maven build."
+  echo ""
+  echo "mvnw clean package > <maven-build-output>"
+  echo ""
+  echo "The environment variable MVN is used to specify the Maven binaries; defaults to 'mvnw'."
+  echo "See further details in the JavaDoc of ShadeOptionalChecker."
+}
+
+while getopts 'h' o; do
+  case "${o}" in
+    h)
+      usage
+      exit 0
+      ;;
+  esac
+done
+
+if [[ "$#" != "1" ]]; then
+  usage
+  exit 1
+fi
+
 ## Checks that all bundled dependencies are marked as optional in the poms
 MVN_CLEAN_COMPILE_OUT=$1
-CI_DIR=$2
-FLINK_ROOT=$3
 
-source "${CI_DIR}/maven-utils.sh"
+MVN=${MVN:-./mvnw}
 
-cd "$FLINK_ROOT" || exit
+dependency_plugin_output=/tmp/optional_dep.txt
 
-dependency_plugin_output=${CI_DIR}/optional_dep.txt
+$MVN dependency:tree -B > "${dependency_plugin_output}"
```

Comment:
```suggestion
$MVN dependency:tree -T1 -B > "${dependency_plugin_output}"
```
That caused the error on my side. I use a local `.mvn/maven.config` to set default parameters (`-DskipTests -Dfast -Pskip-webui-build -T1C` and a custom settings xml that points to a separate local repository). The multi-threading is the problem here. `dependency:tree` can run in multiple threads. But that will screw up the output of course which causes the dependency tree parsing to malfunction :facepalm:

Reply:
I'm wondering whether we should apply this in every mvn command for which we analyze the output. To be on the safe side :thinking:

Reply:
_every_ call (apart from `mvn exec` has to run with a parallelism of 1.